### PR TITLE
Update to Akka.Persistence 1.1.1

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
- Updated to Akka.Persistence 1.1.1
- Added copyrights
- Journal and Snapshot plugin could work separately, now.
- Also fixed https://github.com/akkadotnet/Akka.Persistence.MongoDB/issues/16

### CODE REVIEW

File: src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs [hunk 1]
```diff
@@ -14,71 +23,95 @@ namespace Akka.Persistence.MongoDb.Journal
...
+            var writeTasks = messages.Select(async message =>
+            {
+                var persistentMessages = ((IImmutableList<IPersistentRepresentation>)message.Payload).ToArray();
+
+                var journalEntries = persistentMessages.Select(ToJournalEntry).ToList();
+                await _collection.InsertManyAsync(journalEntries);
+            });
+
+            // Inserting the highest sequence number
+            var persistenceId = messages.Select(c => c.PersistenceId).First();
```
Comment (alexvaluyskiy) [inline src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs]: Probably we should optimize this place

File: src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs [hunk 2]
```diff
@@ -14,71 +23,95 @@ namespace Akka.Persistence.MongoDb.Journal
...
+        protected override Task DeleteMessagesToAsync(string persistenceId, long toSequenceNr)
         {
             var builder = Builders<JournalEntry>.Filter;
             var filter = builder.Eq(x => x.PersistenceId, persistenceId);
-            var update = Builders<JournalEntry>.Update.Set(x => x.IsDeleted, true);
 
             if (toSequenceNr != long.MaxValue)
                 filter &= builder.Lte(x => x.SequenceNr, toSequenceNr);
 
-            if (isPermanent)
```
Comment (alexvaluyskiy) [inline src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs]: Removed permanent delete

**Comment by alexvaluyskiy** [comment]
I have some troubles with tests. They work fine in Visual Studio. But when I run "build RunTests", the script tries to restore the latest dependencies. Like "Mongo2go 1.0.0-beta" and the latest "Immutable.Collections", and tests fail

File: src/Akka.Persistence.MongoDb.Tests/MongoDbSettingsSpec.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+using Akka.Actor;
+using Akka.Util.Internal;
+using FluentAssertions;
+using Xunit;
+
+namespace Akka.Persistence.MongoDb.Tests
+{
+    [Collection("MongoDbSpec")]
+    public class RedisSettingsSpec : Akka.TestKit.Xunit2.TestKit
```
Comment (alexvaluyskiy) [inline src/Akka.Persistence.MongoDb.Tests/MongoDbSettingsSpec.cs]: I should rename it to redis

File: build.fsx [hunk 4]
```diff
@@ -169,8 +173,8 @@ let updateNugetPackages _ =
 
   let getPackages project =
     match project with
-    | "Akka.Persistence.MongoDb" -> ["Akka.Persistence";]
-    | "Akka.Persistence.MongoDb.Tests" -> ["Akka.Persistence.TestKit";]
+    | "Akka.Persistence.MongoDB" -> ["Akka.Persistence";"Akka.Persistence.Sql.Common"]
+    | "Akka.Persistence.MongoDB.Tests" -> ["Akka.Persistence.TestKit";"Akka.Persistence.Sql.Common";]
```
Comment (Silv3rcircl3) [inline build.fsx]: Why do we need Sql.Common here?

File: src/Akka.Persistence.MongoDb.Tests/Akka.Persistence.MongoDb.Tests.csproj [hunk 5]
```diff
@@ -35,52 +35,66 @@
...
       <Private>True</Private>
     </Reference>
-    <Reference Include="MongoDB.Driver.Core, Version=2.0.1.27, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\MongoDB.Driver.Core.2.0.1\lib\net45\MongoDB.Driver.Core.dll</HintPath>
+    <Reference Include="MongoDB.Driver.Core, Version=2.2.4.26, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\MongoDB.Driver.Core.2.2.4\lib\net45\MongoDB.Driver.Core.dll</HintPath>
       <Private>True</Private>
     </Reference>
     <Reference Include="Newtonsoft.Json">
       <HintPath>..\Packages\Newtonsoft.Json.7.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (Silv3rcircl3) [inline src/Akka.Persistence.MongoDb.Tests/Akka.Persistence.MongoDb.Tests.csproj]: that needs to be changed to the new version

File: src/Akka.Persistence.MongoDb/Akka.Persistence.MongoDb.csproj [hunk 6]
```diff
@@ -32,37 +32,42 @@
...
       <Private>True</Private>
     </Reference>
-    <Reference Include="MongoDB.Driver.Core, Version=2.0.1.27, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\packages\MongoDB.Driver.Core.2.0.1\lib\net45\MongoDB.Driver.Core.dll</HintPath>
+    <Reference Include="MongoDB.Driver.Core, Version=2.2.4.26, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\MongoDB.Driver.Core.2.2.4\lib\net45\MongoDB.Driver.Core.dll</HintPath>
       <Private>True</Private>
     </Reference>
     <Reference Include="Newtonsoft.Json">
       <HintPath>..\Packages\Newtonsoft.Json.7.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (Silv3rcircl3) [inline src/Akka.Persistence.MongoDb/Akka.Persistence.MongoDb.csproj]: that needs to be changed to the new version

**Comment by Silv3rcircl3** [comment]
I think we also need dependency redirects in the app.config's because Akka is still using older versions of Json and ImmutableCollections

**Comment by Silv3rcircl3** [comment]
Rest LGTM

File: build.fsx (hunk 4 above)
Comment (alexvaluyskiy) [inline build.fsx]: Oops

File: src/Akka.Persistence.MongoDb.Tests/Akka.Persistence.MongoDb.Tests.csproj (hunk 5 above)
Comment (alexvaluyskiy) [inline src/Akka.Persistence.MongoDb.Tests/Akka.Persistence.MongoDb.Tests.csproj]: Should I update Json.Net to 9.0.1? We should do it first in Akka

**Pull request merged**